application-task-ui/src/main/resources/TaskManager/GanttMacro/JSONTaskSource.xml

```diff
@@ -0,0 +1,195 @@
+    #end
+    #if ("$!request.reporters" != '')
+      #set ($reporters = {})
+      #csvToArray ($request.reporters $reporters)
+      #set ($reporters = $reporters['csvReturn'])
+      #set ($query = $query + " and task.reporter in (:reporters)")
+    #end
+    #if ("$!request.projects" != '')
+      #set ($projects = {})
+      #csvToArray ($request.projects $projects)
+      #set ($projects = $projects['csvReturn'])
+      #set ($query = $query + " and task.project in (:projects)")
+    #end
+    #if ("$!request.from" != '')
+      #set ($from = $datetool.toDate($numbertool.toNumber($request.from).longValue()))
+      #set ($query = $query + ' and (' +
+          '(task.startDate is not NULL and task.startDate  &gt;= :from) or ' +
+          '(task.startDate is     NULL and task.createDate &gt;= :from))'
+      )
+    #end
+    #if ($request.hideNoDueDate != 'false')
+      #set ($query = $query + ' and task.duedate is not NULL')
+    #end
+    #if ("$!request.to" != '')
+      #set ($to = $datetool.toDate($numbertool.toNumber($request.to).longValue()))
+      #set ($query = $query + ' and (' +
+              '(task.duedate is not NULL and task.duedate &lt;= :to) or ' +
+              '(task.duedate is     NULL and (' +
+                '(task.startDate is not NULL and task.startDate  &lt;= :to) or ' +
+                '(task.startDate is     NULL and task.createDate &lt;= :to))' +
+            '))'
+      )
+    #end
+    ## Sort the tasks by project and name to ensure a stable order.
+    #set ($query = $query + ' order by UPPER(task.project), task.project, UPPER(task.name), task.name')
+    #set ($query = $services.query.xwql($query).setLimit(50))
+    #if ($spaces)
+      #set ($query = $query.bindValue("spaces", $spaces))
+      #set ($index = 0)
+      #foreach ($space in $spaces)
+        #set ($space = $space + '.%')
+        #set ($query = $query.bindValue("spaces${index}", $space))
+        #set ($index = $index + 1)
+      #end
+    #end
+    #if ($assignees) #set ($query = $query.bindValue('assignees', $assignees)) #end
+    #if ($projects)  #set ($query = $query.bindValue('projects', $projects))   #end
+    #if ($reporters) #set ($query = $query.bindValue('reporters', $reporters)) #end
+    #if ($from)      #set ($query = $query.bindValue('from', $from))           #end
+    #if ($to)        #set ($query = $query.bindValue('to', $to))               #end
+  #end
+  #set ($queryResult = $query.execute())
+
+  #set ($tasks = [])
+  #foreach ($taskDocument in $queryResult)
+    #set ($taskXDoc = $xwiki.getDocument($taskDocument))
+    #set ($task = $taskXDoc.getObject('TaskManager.TaskManagerClass'))
+    #if ($services.security.authorization.hasAccess('view', $taskDocument))
+      #set ($taskObj = {})
+      #set ($customClasses = '')
+      #set ($discard = $taskObj.put('id', "$taskXDoc.getDocumentReference()"))
+      #set ($discard = $taskObj.put('link', $taskXDoc.getURL()))
+      #putOrDefault ($taskObj 'actualName'
```

Comment:
You cannot use `name` directly?

Reply:
```name``` is a field also used by the Gantt library.

The only use for having 2 name parameters is for removing a warning from the displayed name when a task with no due date is modified (only happens in the ```formatTask``` function in js).